docs/architecture/airbyte-specification.md

```diff
@@ -0,0 +1,180 @@
+            ```
+            {
+              "streams": [
+                {
+                  "name": "customers",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      }
+                    }
+                  }
+                },
+                {
+                  "name": "products",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name", "features"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      },
+                      "features": {
+                        "type": "array",
+                        "items": {
+                          "type": "object",
+                          "required": ["name", "productId"],
+                          "properties": {
+                            "name": "string",
+                            "productId": "number"
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+            ```
+#### Read
+* Input:
+    1. config: A configuration json object that has been validated using the `ConnectorSpecification`.
+    1. catalog: An `AirbyteCatalog`. This catalog should be a subset of the catalog returned by the `discover` command.
+    1. state: A json object. This object is only ever written or read by the source, so it is a json blob with whatever information is necessary to keep track of how much of the data source has already been read. Because, Airbyte currently only supports [Full Refresh](full-refresh.md), this state object is purely cosmetic. It will become more important when Airbyte beings to supporting incremental syncs.
+* Output:
+    1. message stream: A stream of `AirbyteRecordMessage`s and `AirbyteStateMessage`s piped to stdout.
+* This command reads data from the underlying data source and converts it into `AirbyteRecordMessage`.
+* Outputting `AirbyteStateMessages` is optional. They can be used to track how much of the data source has been synced.
+* The integration ideally will only pull the data described in the catalog argument. It is permissible for the integration, however, to ignore the catalog and pull data from any stream it can find. If it follows this second behavior, the extra data will be pruned in the worker. We prefer the former behavior because it reduces the amount of data that is transferred and allows control over not sending sensitive data. There are some sources for which this is simply not possible.
+
+### Destination
+* A source is implemented as a docker container. The container must adhere to the following interface.
+```
```

Comment:
Same for the ordering. I would recommend showing the full CLI example here, even if you skip going into detail for the repeats.

Reply:
ok